src/HealthChecks.RabbitMQ/DependencyInjection/RabbitMQHealthCheckBuilderExtensions.cs

```diff
@@ -33,5 +33,92 @@ public static IHealthChecksBuilder AddRabbitMQ(this IHealthChecksBuilder builder
                 tags,
                 timeout));
         }
+
+        /// <summary>
+        /// Add a health check for RabbitMQ services using <see cref="IConnection"/> from service provider 
+        /// or <see cref="IConnectionFactory"/> from service provider if none is found. At least one must be configured.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'rabbitmq' will be used for the name.</param>
+        /// <param name="failureStatus">
+        /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+        /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+        /// </param>
+        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+        /// <param name="timeout">An optional System.TimeSpan representing the timeout of the check.</param>
+        /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns></param>
+        public static IHealthChecksBuilder AddRabbitMQ(this IHealthChecksBuilder builder, string name = default, 

```

Comment:
One problem here is that there is ambiguity between the existing method's signature and this one if you are only specifying a single string. I can either rename this method or consolidate into a single method where it has different behavior depending on what fields are provided to prevent this ambiguity.

Reply:
We could resolve the ambiguity by re-ordering the parameters here, but that feels a little cheaty:

```csharp
public static IHealthChecksBuilder AddRabbitMQ(this IHealthChecksBuilder builder, HealthStatus? failureStatus = default, string name = default, 
```

I think with that ordering a call like `AddRabbitMQ("amqps://broker")` would resolve to the overload on L27, and if we wanted to call this overload we'd need something like `AddRabbitMQ(name: "broker")`

Reply:
The thought I had today was to add a bool indicating whether to give priority to IConnectionFactory instead of IConnection. If I make it optional. I introduce a possibly useful feature and I maintain backwards compatibility.

Reply:
that'd do it; what would that parameter be called? `preferIConnectionFactoryFromServices` ?

the code might get a little messy to make this work as a priority; how about making it a little narrower, something like: `bool useIConnectionFactoryFromServices=false`. If it's true, we only look for an `IConnectionFactory`, if it's false, we only look for an `IConnection`?